src/Xenial.Tasty.Cli/TastyServer.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Protocols;
+using Xenial.Delicious.Reporters;
+using Xenial.Delicious.Utils;
+
+using static Xenial.Delicious.Utils.Actions;
+
+namespace Xenial.Delicious.Cli
+{
+    public class TastyServer
+    {
+        public event EventHandler<ExecuteCommandEventArgs>? ExecuteCommand;
+        public event EventHandler? CancellationRequested;
+
+        internal async Task DoExecuteCommand(ExecuteCommandEventArgs args)
+        {
+            ExecuteCommand?.Invoke(this, args);
+            await Task.CompletedTask;
+        }
+
+        internal async Task DoRequestCancellation()
+        {
+            CancellationRequested?.Invoke(this, EventArgs.Empty);
+            await Task.CompletedTask;
+        }
+
+        public Task Report(SerializableTestCase @case)
+            => ConsoleReporter.Report(@case);
+
+        public Task Report(IEnumerable<SerializableTestCase> @cases)
+           => ConsoleReporter.ReportSummary(@cases);
+
+        internal Action<IList<SerializableTastyCommand>>? CommandsRegistered;
```

Comment:
A weird mix of null and null implementations on these actions

Reply:
Again JsonRpc conventions

Reply:
I hate the fact that JsonRPC is not really good at throwing awaitable streams here, proably we should look more into async enumerable streams, but that will will any full framework support, which is yet to be decided to be supported for 1.0